### ISSUE TITLE
Install fails on CiviCRM 4.7.20+

### ISSUE THREAD

**Issue by seancolsen** [mlutfy/ca.bidon.pcpteams#9]
Attempting to install this extension on CiviCRM 4.7.20 or later fails with the following error:

> CRM_Exception: Could not find the SQL file. in /home/sean/buildkit/build/ext/sites/default/files/civicrm/ext/ca.bidon.pcpteams/CRM/Pcpteams/Upgrader/Base.php on line 104

**Comment by seanmadsen** [comment]
I've been trying to work on a fix, but it's a bit of a pickle! 

In https://github.com/civicrm/civicrm-core/commit/c0e4c31d33e2bc7f7b53891f5b6481b719bed1d8 / https://github.com/civicrm/civicrm-core/pull/10216 (CiviCRM 4.7.20) the function signature of `CRM_Utils_File::sourceSQLFile` changed so that it no longer accepts raw SQL. That's a bummer because this extension uses that featre [here](https://github.com/seanmadsen/ca.bidon.pcpteams/blob/ec382b7682a50381ab3617b246bb108e0fa6d51a/CRM/Pcpteams/Upgrader/Base.php#L100).

This patch fixes PCP Teams so that it installs on 4.7.20+ 

```patch
Index: CRM/Pcpteams/Upgrader/Base.php
===================================================================
--- CRM/Pcpteams/Upgrader/Base.php	(date 1500611626000)
+++ CRM/Pcpteams/Upgrader/Base.php	(date 1505510138000)
@@ -97,11 +97,9 @@
         $smarty->template_dir = array($extRoot . '../../../', $smarty->template_dir);
       }
 
-      CRM_Utils_File::sourceSQLFile(
+      CRM_Utils_File::runSqlQuery(
         CIVICRM_DSN,
-        $smarty->fetch($this->extensionDir . '/' . $relativePath),
-        NULL,
-        TRUE
+        $smarty->fetch($this->extensionDir . '/' . $relativePath)
       );
     }
     else {
```

But to further complicate matters, that file which needs to be changed, is *auto-generated* from civix. Maybe we can just re-generate it? I'm not sure if (a) that would even fix it and (b) if re-generating the file would break other things on 4.6. 

Any thoughts @mlutfy? 

On a related note you can see [here](https://github.com/civicrm/civicrm-core/commit/c0e4c31d33e2bc7f7b53891f5b6481b719bed1d8#commitcomment-23332432) that another extension failed to install after this change to core :confused: But it looks like the [fix for that](https://github.com/Project60/org.project60.banking/pull/168/commits/324edd467f358f53bd71f79859e4c8705be481e7) was a bit simpler because it didn't actually need SQL rendered with smarty.

**Comment by mlutfy** [comment]
This has finally been fixed (yikes, 3 years later!). Thank you Sean!

The underlying problematic code has been removed from the extension since CiviCRM core now provides PCP notifications.
